From 89460f3fc3d6c0659d44897a6adec3c3fecc5564 Mon Sep 17 00:00:00 2001
Message-Id: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Fri, 23 Jan 2015 12:43:39 +0000
Subject: [PATCH 01/68] Revert "FOR_UPSTREAM [VPG]: drm/i915: Fix PPGTT PDP
 allocations"

This reverts commit ed44bda7f9a026ef9e9a3d92dea222dabc9fb39c.

Android code was pointing the unused PDPs to a scratch page to prevent
page faults.

Dynamic page allocation will do the same, so revert this patch to allow
a cleaner merge from upstream.

Change-Id: Ie34576b4d8d50acd415ecca6e424e7c0e4ffb902
For: GMINL-5449
For: VIZ-4619
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index 94ec541..9333698 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -603,7 +603,7 @@ static int gen8_ppgtt_init(struct i915_hw_ppgtt *ppgtt, uint64_t size)
 		DRM_INFO("Pages will be wasted unless GTT size (%llu) is divisible by 1GB\n", size);
 
 	/* 1. Do all our allocations for page directories and page tables. */
-	ret = gen8_ppgtt_alloc(ppgtt, GEN8_LEGACY_PDPS);
+	ret = gen8_ppgtt_alloc(ppgtt, max_pdp);
 	if (ret)
 		return ret;
 
-- 
1.7.9.5

